### DISCUSSION TITLE
[New Feature Request]: Add rlwrap support to OpenSTA shell

### DISCUSSION THREAD

**Discussion by arunkpv** [parallaxsw/OpenSTA#23] [Ideas]
Hello!
I have started using OpenSTA only for quite a few months.

- To try out the various commands and learn their usage, I had to use the OpenSTA interactive shell.
- The interactive shell usage was also used when I had to do analysis of certain STA failures manually.

- But it was a pain to use the shell as there is no command history scroll and the irritating `^[[D`, `^[[A`, etc. that you see when pressing the arrow keys.
- Searching for solutions online, I came across the **rlwrap** utility and I am currently invoking OpenSTA from the terminal with rlwrap: `rlwrap sta`
  **rlwrap GitHub link:** [https://github.com/hanslub42/rlwrap.git](https://github.com/hanslub42/rlwrap.git)

Would it be possible to add defacto/ by default support of rlwrap by packaging it into the OpenSTA binary so that the interactive shell usage becomes more user-friendly ?
Making this request since there are use cases like invoking the OpenSTA from within the OpenLane docker images, where it does not seem possible to install rlwrap (or any packages) using apt-get.

**Comment by maliberty** [comment]
I believe you want tclreadline, see the option USE_TCL_READLINE https://github.com/The-OpenROAD-Project/OpenSTA/blob/a7f34210b403fe399c170296d54258f10f92885f/CMakeLists.txt#L33

**Comment by arunkpv** [comment]
Hi @maliberty,
Oh okay! 
That looks like a compile-time variable, right ?
I feel dumb now that you have pointed it out; I somehow missed that during my local installation.

However, I still have the problem with the OpenSTA build bundled through OpenLane docker image, where I am unable to do anything.
Wouldn't it be better to set USE_TCL_READLINE set to ON by default ?

**Comment by jjcherry56** [comment]
Leaving it OFF minimizes compile dependencies, which are always problematic.
That is why all optional dependencies are off by default.
You can always run it inside an emacs shell window and get full editing.

**Comment by arunkpv** [comment]
Understood.
Closing this discussion.
